File: memory-domain/src/main/java/com/memory/domain/diary/Diary.java
package com.memory.domain.diary;

import com.memory.domain.BaseTimeEntity;
import com.memory.domain.member.Member;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@ToString
@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Diary extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    private LocalDate date;

    private String mood;

    private String weather;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member member;

    @Builder
    private Diary(String title, String content, LocalDate date, String mood, String weather, Member member) {
        this.title = title;
        this.content = content;
        this.date = date;
        this.mood = mood;
        this.weather = weather;
        this.member = member;
    }

    public static Diary create(String title, String content, LocalDate date, String mood, String weather, Member member) {
        return new Diary(title, content, date, mood, weather, member);
    }

    public void update(String title, String content, LocalDate date, String mood, String weather) {
        this.title = title;
        this.content = content;
        this.date = date;
        this.mood = mood;
        this.weather = weather;
    }

    public boolean isOwner(Member member) {
        return !this.member.getId().equals(member.getId());
    }
}

File: memory-domain/src/main/kotlin/com/memory/domain/diary/Diary.kt
package com.memory.domain.diary

import com.memory.domain.BaseTimeEntity
import com.memory.domain.member.Member
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Diary(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    var title: String? = null,

    @Column(columnDefinition = "TEXT")
    var content: String? = null,

    var date: LocalDate? = null,

    var mood: String? = null,

    var weather: String? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    var member: Member? = null,
) : BaseTimeEntity() {

    companion object {
        @JvmStatic
        fun create(
            title: String,
            content: String?,
            date: LocalDate,
            mood: String?,
            weather: String?,
            member: Member
        ): Diary = Diary(
            title = title,
            content = content,
            date = date,
            mood = mood,
            weather = weather,
            member = member
        )
    }

    fun update(
        title: String,
        content: String,
        date: LocalDate,
        mood: String?,
        weather: String?
    ) {
        this.title = title
        this.content = content
        this.date = date
        this.mood = mood
        this.weather = weather
    }

    fun isOwner(member: Member): Boolean =
        this.member?.id != member.id
}